pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/errorprone/xml/MissingEqualsOnComparable.xml

```diff
@@ -0,0 +1,281 @@
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            public class ConcreteClass {
+                public boolean equals(Object o) {
+                    return false;
+                }
+            }
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>enum declaration with no methods</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            public enum SimpleEnum {
+                VALUE1, VALUE2
+            }
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>enum declaration with equals only</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            public enum EnumWithEquals {
+                VALUE1, VALUE2;
+
+                @Override
+                public boolean equals(Object o) {
+                    return false;
+                }
+            }
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>enum declaration with hashCode only</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            public enum EnumWithHashCode {
+                VALUE1, VALUE2;
+
+                @Override
+                public int hashCode() {
+                    return 42;
+                }
+            }
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>enum declaration with both equals and hashCode</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            public enum CompleteEnum {
+                VALUE1, VALUE2;
+
+                @Override
+                public boolean equals(Object o) {
+                    return false;
+                }
+
+                @Override
+                public int hashCode() {
+                    return 42;
+                }
+            }
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>enum implementing Comparable with equals</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+            public enum ComparableEnum implements Comparable<ComparableEnum> {
+                VALUE1, VALUE2;
+
+                @Override
+                public boolean equals(Object o) {
+                    return false;
+                }
+            }
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>Non-Comparable class with equals but no hashCode</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+        class Foo {
+            @Override
+            public boolean equals(Object o) {
+                return false;
+            }
+        }
+    ]]></code>
+    </test-code>
+    <test-code>
+        <description>Record with Comparable and no explicit equals/hashCode</description>
+        <expected-problems>0</expected-problems>
```

Comment:
As explained in the other comments, I think we should actually report records here...

Reply:
done thx.